xmppserver/src/test/java/org/jivesoftware/util/CertificateManagerTest.java

```diff
@@ -272,23 +288,28 @@ public void testServerIdentitiesXmppAddrAndDNS() throws Exception
         final String subjectAltNameDNS = "MySubjectAltNameDNS";
 
         final X509v3CertificateBuilder builder = new JcaX509v3CertificateBuilder(
-                new X500Name( "CN=MyIssuer" ),                                          // Issuer
-                BigInteger.valueOf( Math.abs( new SecureRandom().nextInt() ) ),         // Random serial number
-                new Date( System.currentTimeMillis() - ( 1000L * 60 * 60 * 24 * 30 ) ), // Not before 30 days ago
-                new Date( System.currentTimeMillis() + ( 1000L * 60 * 60 * 24 * 99 ) ), // Not after 99 days from now
-                new X500Name( "CN=" + subjectCommonName ),                              // Subject
+                new X500Name( "CN=MyIssuer" ),                          // Issuer
+                BigInteger.valueOf( Math.abs( new SecureRandom().nextInt() ) ), // Random serial number
+                Date.from( Instant.now().plus(Duration.ofDays(30)) ),           // Not before 30 days ago
+                Date.from( Instant.now().minus(Duration.ofDays(99)) ),          // Not after 99 days from now
+                new X500Name( "CN=" + subjectCommonName ),              // Subject
                 subjectKeyPair.getPublic()
         );
 
-        final DERSequence otherName = new DERSequence( new ASN1Encodable[] { XMPP_ADDR_OID, new DERUTF8String( subjectAltNameXmppAddr ) });
+        final OtherName otherName = new OtherName(XMPP_ADDR_OID, new DERUTF8String( subjectAltNameXmppAddr ) );
         final GeneralNames subjectAltNames = new GeneralNames( new GeneralName[] {
                 new GeneralName( GeneralName.otherName, otherName ),
                 new GeneralName( GeneralName.dNSName, subjectAltNameDNS )
         });
         builder.addExtension( Extension.subjectAlternativeName, true, subjectAltNames );
 
         final X509CertificateHolder certificateHolder = builder.build( contentSigner );
-        final X509Certificate cert = new JcaX509CertificateConverter().getCertificate( certificateHolder );
+        X509Certificate cert = new JcaX509CertificateConverter().getCertificate( certificateHolder );
+
+        // FIXME: Unsure why, but without this back-and-forth, tests will fail on Java 17.
```

Comment:
I'm not sure this *is* a "FIXME", as such - if it is indeed some kind of error, it's one in the test code and not the code under test.

Reply:
I agree that it's likely the test code. I'd like that to be fixed too, though.

Reply:
Fair point.